### PR TITLE
chore: fix hybrid models

### PR DESCRIPTION
**Pull request by jfrery**
closes https://github.com/zama-ai/concrete-ml-internal/issues/3905

cc @RomanBredehoft

### CODE REVIEW

**Review by RomanBredehoft — DISMISSED** [review]
thanks !

**Review by fd0r — CHANGES_REQUESTED** [review]
The divergence in results was between torch and quantized?

I would still keep a torch mode here is instead of removing it entirely.

**Comment by jfrery** [comment]
> The divergence in results was between torch and quantized?
> 
> I would still keep a torch mode here is instead of removing it entirely.

Not sure what you mean. I check the accuracy of simulation vs torch and disable vs torch.

Why would we keep a torch execution? We have the torch model for this

File: src/concrete/ml/torch/hybrid_model.py [hunk 1]
```diff
@@ -519,12 +508,17 @@ def _save_fhe_circuit(self, path: Path, via_mlir=False):
 
         model_path = Path(path)
         for module_name in self.module_names:
+            onnx_model = self.private_q_modules[  # pylint: disable=protected-access
```
Comment (RomanBredehoft) [inline src/concrete/ml/torch/hybrid_model.py:511]: there is not need to access the private `_onnx_model`, `QuantizedModule` instances have a public property for them , you can directly call `onnx_model` normally

Comment (jfrery) [reply src/concrete/ml/torch/hybrid_model.py:511]: Ah yes!

**Comment by RomanBredehoft** [comment]
actually @jfrery , I think @fd0r is right we should keep the torch execution as well, but maybe through a new mode like `"torch"` or `"float"` ? 

because as I mentioned to you, currently the torch model is modified in place by the hybrid model. But that means, there's no real way of running the model in float after compiling it with the hybrid

**Comment by jfrery** [comment]
> actually @jfrery , I think @fd0r is right we should keep the torch execution as well, but maybe through a new mode like `"torch"` or `"float"` ?
> 
> because as I mentioned to you, currently the torch model is modified in place by the hybrid model. But that means, there's no real way of running the model in float after compiling it with the hybrid

Very good point! We replace in-place so I am basically checking disable vs disable and simulate vs disable. I will add the torch inference.

That being said, simulate vs disable should raise the flaky we saw ...

File: src/concrete/ml/torch/hybrid_model.py (hunk 1 above)
Comment (jfrery) [reply src/concrete/ml/torch/hybrid_model.py]: Fixed

File: tests/torch/test_hybrid_converter.py [hunk 2]
```diff
@@ -127,9 +126,9 @@ def run_hybrid_llm_test(
 @pytest.mark.parametrize(
     "list_or_str_private_modules_names, expected_accuracy, has_pbs",
     [
-        ("transformer.h.0.mlp", 0.934, True),
-        (["transformer.h.0.mlp", "transformer.h.1.mlp"], 0.42, True),
-        ("transformer.h.0.mlp.c_fc", 0.986, False),
+        ("transformer.h.0.mlp", 0.95, True),
+        (["transformer.h.0.mlp", "transformer.h.1.mlp"], 0.40, True),
```
Comment (RomanBredehoft) [inline tests/torch/test_hybrid_converter.py:130]: do we have an idea why this accuracy is so low then @jfrery ?

**Review by RomanBredehoft — APPROVED** [review]
thanks ! maybe wait for @fd0r feedback as well

File: tests/torch/test_hybrid_converter.py (hunk 2 above)
Comment (jfrery) [reply tests/torch/test_hybrid_converter.py:130]: The "accuracy" here is the percent of similar labels in the top5 for all examples so a bit specific and I am not sure what to expect.

I think the rounding is really bad on MLP due to the outliers observed in LLMs. We might want to try smoothquant on gpt2 to see if this makes the rounding a bit less aggressive.

**Review by fd0r — APPROVED** [review]
LGTM

**Pull request merged**